src/lib/y2network/sysconfig/connection_config_writers/tap.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/sysconfig/connection_config_writers/tun"
+
+module Y2Network
+  module Sysconfig
+    module ConnectionConfigWriters
+      # This class is responsible for writing the information from a ConnectionConfig::Tap
+      # object to the underlying system.
+      class Tap < Tun
```

Comment:
i think that inheritance should somehow reflect real world. Since Tap is not any extension to tun and both technologies work even on different layers of Iso / osi model, I don't think that this direction of inheritance is correct.

Reply:
But that is some underlying thing. I mean, for us it only differs in the 'mode' that is the 'tunnel' attribute.

Even iproute2 uses `ip tuntap ...`

It was basically for DRY. Originally was even thinking in just add it as tuntap and the mode, but preferred to separate the types as we had for InterfaceType.

Reply:
SO, if you prefer will not inherit the connection config.

Reply:
@imobachgs @mchf 

For the reader & writer, I would keep at it is, but as you prefer

Reply:
Also removed the inheritance.